framework/src/constraints/AutomaticMortarGeneration.C

```diff
@@ -133,22 +159,30 @@ AutomaticMortarGeneration::buildMortarSegmentMesh()
 
     std::vector<Node *> new_nodes;
     for (MooseIndex(secondary_elem->n_nodes()) n = 0; n < secondary_elem->n_nodes(); ++n)
-      new_nodes.push_back(
-          mortar_segment_mesh.add_point(secondary_elem->point(n), secondary_elem->node_id(n)));
+    {
+      new_nodes.push_back(mortar_segment_mesh->add_point(
+          secondary_elem->point(n), secondary_elem->node_id(n), secondary_elem->processor_id()));
+      Node * const new_node = new_nodes.back();
+      new_node->set_unique_id(new_node->id() + node_unique_id_offset);
+    }
 
     Elem * new_elem;
     if (secondary_elem->default_order() == SECOND)
-      new_elem = mortar_segment_mesh.add_elem(new Edge3);
+      new_elem = new Edge3;
     else
-      new_elem = mortar_segment_mesh.add_elem(new Edge2);
+      new_elem = new Edge2;
```

Comment:
Mortar mesh is always 1D?

Reply:
Yea we do not do mortar with 3D (e.g. 2D mortar segment mesh elements) yet